pkg/build/builder/dockerutil.go

```diff
@@ -156,15 +156,27 @@ func dockerRun(client DockerClient, createOpts docker.CreateContainerOptions, lo
 
 	containerName := containerNameOrID(c)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a sigterm
```

Comment:
It is design exactly fro this use case.  Put your remove in your defer inside run(func() {}).  No channels or signals needed.